runtime/runtime.go

```diff
@@ -644,20 +716,73 @@ func (r *interpreterRuntime) parseAndCheckProgram(
 		return nil, wrapError(err)
 	}
 
+	// NOTE: cache elaboration *before* checking,
+	// so it is returned when there is a cyclic import
+
+	elaboration = checker.Elaboration
+
+	wrapPanic(func() {
+		err = startContext.Interface.CacheElaboration(startContext.Location, elaboration)
+	})
+	if err != nil {
+		return nil, wrapError(err)
+	}
+
 	err = checker.Check()
 	if err != nil {
 		return nil, wrapError(err)
 	}
 
-	return checker, nil
+	return elaboration, nil
+}
+
+func (r *interpreterRuntime) checkLocation(
+	context Context,
+	functions stdlib.StandardLibraryFunctions,
+	options []sema.Option,
+	useCache bool,
+) (
+	*interpreter.Program,
+	error,
+) {
+	var err error
+	var program *ast.Program
+	program, err = r.parseLocation(context, useCache)
+	if err != nil {
+		return nil, err
+	}
+
+	var elaboration *sema.Elaboration
+
+	if useCache {
+		wrapPanic(func() {
+			elaboration, err = context.Interface.GetCachedElaboration(context.Location)
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if elaboration == nil {
+		elaboration, err = r.check(program, context, functions, options, useCache)
```

Comment:
just a minor comment that might be more readable if the cacheElaboration is here than inside the check, thoughts ?

Reply:
I think you commented on older code, this function is gone in the last commit. Maybe refresh to make sure you're looking at the latest code